### PR TITLE
Update ebpf documentation

### PR DESCRIPTION
**Pull request by thiagoftsm**
##### Summary
This PR is the first step to improve our documentation, it is bringing updates for `VFS`, `Process` and `Socket`, it also adds contents for `System Overview`, `Memory`, `Disk` and `File system`.

##### Component Name
eBPF.plugin Documentation.
##### Test Plan
No test needed, we are only updating documentation.

### CODE REVIEW

File: collectors/ebpf.plugin/README.md
Comment (hugovalente-pm) [inline collectors/ebpf.plugin/README.md:90]:
```suggestion
#### Synchronization
```

File: collectors/ebpf.plugin/README.md
Comment (hugovalente-pm) [inline collectors/ebpf.plugin/README.md:83]:
```suggestion
chart uses data from the four functions we are monitoring, mentioned above.
```

File: collectors/ebpf.plugin/README.md
Comment (hugovalente-pm) [inline collectors/ebpf.plugin/README.md:123]:
```suggestion
This group has two charts demonstrating how applications interact with the Linux kernel to open and close file descriptors.
```

File: collectors/ebpf.plugin/README.md
Comment (hugovalente-pm) [inline collectors/ebpf.plugin/README.md:167]: Would be good to link this to the line 90 (#syncrhonization) but I'm not sure how this will work, since we have two sub-sections with Synchronization name...

**Review by hugovalente-pm — CHANGES_REQUESTED** [review]
Please also take the opportunity to correct this on line 287 https://github.com/netdata/netdata/pull/11440/files#diff-55ed244c83999827bc3c6f49ad81b31fb70b31215027a9e4646f356095c0eaadR287 - change `process` to `processes`

File: collectors/ebpf.plugin/README.md
Comment (underhood) [inline collectors/ebpf.plugin/README.md:50]: I think this sentence is not clear and could be understood (at least by me) in 2 ways:
- All charts are disabled by default
- Not all of the charts are enabled by default (some are)

Comment (ilyam8) [reply collectors/ebpf.plugin/README.md:50]: "...consume a lot of resources" sounds scary, i would never enable all the charts 😄

File: collectors/ebpf.plugin/README.md
Comment (underhood) [inline collectors/ebpf.plugin/README.md:43]: should menu names be enclosed in `backticks` (``)? we do that for chart names below

File: collectors/ebpf.plugin/README.md
Comment (underhood) [inline collectors/ebpf.plugin/README.md:92]: Either `The **synchronization** submenu OF Netdata monitors calls...` or `IN the **synchronization** submenu, Netdata monitors...`

Comment (underhood) [reply collectors/ebpf.plugin/README.md:92]: also `calls for different syscalls` sounds confusing to me. Probably would put it as `monitors usage of various kernel syscalls`

File: collectors/ebpf.plugin/README.md
Comment (underhood) [inline collectors/ebpf.plugin/README.md:97]:
```suggestion
This chart shows calls to synchronization methods, `fsync()` and `fdatasync()`, to transfer all modified page caches for the 
```

File: collectors/ebpf.plugin/README.md
Comment (underhood) [inline collectors/ebpf.plugin/README.md:101]: Should probably unify capitalization of captions? We have
- Memory map sync (Mms)
and below it
- File System sync
e.g. every caption seems to use different rules

**Comment by ilyam8** [comment]
I find it hard to see the hierarchy

It is
- Memory
  - Page cache
    - Page cache ratio
    - Dirty pages
    - Page cache hits
    - Page cache misses

That is how it looks on GitHub

Screenshot 2021-08-16 at 18 58 00

---

If we don't need named anchors for every chart, then we could simplify the document using lists or tables. It would decrease README length too.

That is just my opinion.

File: collectors/ebpf.plugin/README.md
Comment (thiagoftsm) [reply collectors/ebpf.plugin/README.md:167]: I agree, let us merge this code like it is and when I bring more updates with other PRs, we create the links.

File: collectors/ebpf.plugin/README.md
Comment (thiagoftsm) [reply collectors/ebpf.plugin/README.md:50]: ping @hugovalente-pm !

Comment (thiagoftsm) [reply collectors/ebpf.plugin/README.md:50]: @ilyam8 when I address the latency areas, one of the features I will bring it will be to extend the time we are reading data from kernel ring. These eBPF programs are on top of heavy operations, this is the motive we brought this warning.

File: collectors/ebpf.plugin/README.md
Comment (thiagoftsm) [reply collectors/ebpf.plugin/README.md:43]: A good point, I thought the bold around would be enough, @hugovalente-pm can you clarify this for us?

**Comment by thiagoftsm** [comment]
> I find it hard to see the hierarchy
> 

Our goal is to create links for each one of the sections, they will be used on our dashboard. We will also extend the content in the next PRs. 

Now about the reorganization, I am not sure how we can have a better organization using this language used to create doc, @hugovalente-pm what do you think?

File: collectors/ebpf.plugin/README.md
Comment (underhood) [reply collectors/ebpf.plugin/README.md:43]: > System Overview, Memory, Disks, Filesystem, Mount Points and Applications

Is neither bold nor inline code (with backticks)

File: collectors/ebpf.plugin/README.md
Comment (underhood) [reply collectors/ebpf.plugin/README.md:50]: > "...consume a lot of resources" sounds scary, i would never enable all the charts

All your CPUz belong to us!

**Comment by hugovalente-pm** [comment]
> > I find it hard to see the hierarchy
> 
> Our goal is to create links for each one of the sections, they will be used on our dashboard. We will also extend the content in the next PRs.
> 
> Now about the reorganization, I am not sure how we can have a better organization using this language used to create doc, @hugovalente-pm what do you think?

As you said we need the anchors to link to our dashboards, @ilyam8 suggestion was to put tables if anchors were not needed.
The only thing I see, but would increase reading length, is to put a bullet list under the Memory section to show what are the sub-sections below:

* Page cache
   * Page cache ratio
   * Dirty pages
   * Page cache hits
   * Page cache misses
* Synchronization
   * File sync
   * Memory map sync
   * File System sync
   * File Range sync

File: collectors/ebpf.plugin/README.md
Comment (hugovalente-pm) [reply collectors/ebpf.plugin/README.md:50]: @thiagoftsm @ilyam8 from what I understood these configurations should only be enabled if someone is trying to troubleshoot and needs to have full visibility into the date we collect with eBPF, right?

I believe the callout is needed, what we could add is an highlight to show in which case a user could want to enabled this and what would be the benefits of doing it

File: collectors/ebpf.plugin/README.md
Comment (hugovalente-pm) [reply collectors/ebpf.plugin/README.md:43]: was trying to find other examples and could only find this one https://learn.netdata.cloud/docs/agent/collectors/apps.plugin where bold is used for **Applications**, **Users** and **User Groups**, but this was the only I could find not sure if there are other examples.

imo bold works ok as long as we do on all, like the place @underhood called, but this is just personal opinion

File: collectors/ebpf.plugin/README.md
Comment (ilyam8) [reply collectors/ebpf.plugin/README.md:50]: I agree that warning is needed. My point is that being more specific is better - what resources (cpu, mem, etc), how many resources.

File: collectors/ebpf.plugin/README.md
Comment (thiagoftsm) [reply collectors/ebpf.plugin/README.md:100]: Thank you to call attention for this, I changed and now only the first letter of the sentence is upper case.

File: collectors/ebpf.plugin/README.md
Comment (UmanShahzad) [inline collectors/ebpf.plugin/README.md:211]:
```diff
-`udp_sendmsg`, and `udp_recvmsg` responsible to send and receive
+`udp_sendmsg` and `udp_recvmsg`, which are responsible for sending & receiving
```

File: collectors/ebpf.plugin/README.md
Comment (UmanShahzad) [inline collectors/ebpf.plugin/README.md:212]:
```diff
-when `UDP` protocol
+when the `UDP` protocol
```

File: collectors/ebpf.plugin/README.md
Comment (UmanShahzad) [inline collectors/ebpf.plugin/README.md:216]:
```diff
-Like previous section,
+Like the previous section,
```

```diff
-`tcp_sendmsg` and `tcp_cleanup_rbuf`but,  instead of
+`tcp_sendmsg` and `tcp_cleanup_rbuf`, but instead of
```

File: collectors/ebpf.plugin/README.md
Comment (thiagoftsm) [reply collectors/ebpf.plugin/README.md]: I added some additional description trying to let it more soft, when the latest version of `libbpf` is merged during this week, we will be able to reduce the overhead, because new features will be available for us, but these features depend of kernel compilation, so we need to keep some warnings like this.

Comment (hugovalente-pm) [reply collectors/ebpf.plugin/README.md]: @thiagoftsm, as @ilyam8  suggedted, can we name and, if possible, measure what are the impacts in terms of resources and a value of % of increase in usage?

Comment (thiagoftsm) [reply collectors/ebpf.plugin/README.md]: @hugovalente-pm it is possible, and I will measure to bring in the next PR, for this PR I would prefer to add this missing documentation for we have the links, in the next PR, that I expect to bring next days, I will add a table with these values.

I accepted @underhood suggestion.

File: collectors/ebpf.plugin/README.md
Comment (thiagoftsm) [reply collectors/ebpf.plugin/README.md]: You understood correctly, I am using your suggestion.

File: collectors/ebpf.plugin/README.md
Comment (thiagoftsm) [reply collectors/ebpf.plugin/README.md:43]: @underhood this is the unique change I did not bring, I will wait for your answer to proceed.

**Comment by thiagoftsm** [comment]
@ilyam8 and @hugovalente-pm  the last commit is removing the additional level, I am not sure whether the result is what you were expecting.I thought to remove the sections `Page cache` and `Synchronization` from memory and add the description to the new bullets, but I would like to hear your opinion.

File: collectors/ebpf.plugin/README.md
Comment (underhood) [reply collectors/ebpf.plugin/README.md:43]: @thiagoftsm up to you then, wont block on this

**Review by UmanShahzad — APPROVED** [review]
Thanks for addressing my comments!

File: collectors/ebpf.plugin/README.md
Comment (ilyam8) [inline collectors/ebpf.plugin/README.md:82]: "Page cache" is a submenu according to the Memory bulleted list, but "Page cache", "Page cache ratio", and "Dirty pages" have the same heading level (number of `#`).

It is not the only place you have that problem, please go through the "Charts" section and fix where is needed.

If i understand you right, you want to use:
- `###` top level menu section
- `####` sub-menu section
- `#####` a chart (title?)

Comment (thiagoftsm) [reply collectors/ebpf.plugin/README.md]: All right, I moved the data from `Page cache` to submenu and I removed the menu, because they were already explained in `Mem`. 

I removed `Synchronization` without to move content, because the `syscalls` monitored are already cited in chart description.

File: collectors/ebpf.plugin/README.md
Comment (thiagoftsm) [reply collectors/ebpf.plugin/README.md:43]: Thank you! I will move forward with @hugovalente-pm 's suggestion, but we will have more PRs related to these group of charts, so we will have opportunity to change them again.

**Review by UmanShahzad — DISMISSED** [review]

File: collectors/ebpf.plugin/README.md
Comment (vlvkobal) [inline collectors/ebpf.plugin/README.md:79]: There are many places with whitespaces at the beginning of strings or at the end, like in these lines. Please remove them.

File: collectors/ebpf.plugin/README.md
Comment (vlvkobal) [inline collectors/ebpf.plugin/README.md:170]:
```suggestion
`fdatasync(2)` on a file. You can see more details in the Synchronization section.
```

File: collectors/ebpf.plugin/README.md
Comment (vlvkobal) [inline collectors/ebpf.plugin/README.md:205]:
```suggestion
Like the previous chart, this one also monitors `tcp_sendmsg` and `tcp_cleanup_rbuf` but, instead of showing the 
```
BTW, why there is the `close` dimension in this chart? Shouldn't it be deleted?

File: collectors/ebpf.plugin/README.md
Comment (vlvkobal) [inline collectors/ebpf.plugin/README.md:220]:
```suggestion
Like the previous chart, this one also monitors `udp_sendmsg` and `udp_recvmsg`, but instead of showing the 
```

File: collectors/ebpf.plugin/README.md
Comment (vlvkobal) [inline collectors/ebpf.plugin/README.md:200]:
```suggestion
This chart demonstrates calls to functions `tcp_sendmsg`, `tcp_cleanup_rbuf`, and `tcp_close`; these functions are
```

File: collectors/ebpf.plugin/README.md
Comment (vlvkobal) [inline collectors/ebpf.plugin/README.md:181]:
```suggestion
system calls: `fork(2)`, `vfork(2)`, and `clone(2)`. In turn, each of these system calls uses either the function
```

File: collectors/ebpf.plugin/README.md
Comment (underhood) [reply collectors/ebpf.plugin/README.md:205]: Also comma here should be before not after "but"

**Comment by hugovalente-pm** [comment]
> @ilyam8 and @hugovalente-pm the last commit is removing the additional level, I am not sure whether the result is what you were expecting.I thought to remove the sections `Page cache` and `Synchronization` from memory and add the description to the new bullets, but I would like to hear your opinion.

@thiagoftsm to be honest I was expecting to introduce the short index to give context to the users of what would be coming ahead and where things fit, but still keep the `Page cache` and `Synchronization`.  I think I would still prefer with the sections but it doesn't shock me if we proceed was you did.

File: collectors/ebpf.plugin/README.md
Comment (UmanShahzad) [reply collectors/ebpf.plugin/README.md:205]: @vlvkobal I believe @underhood is right.

File: collectors/ebpf.plugin/README.md
Comment (UmanShahzad) [reply collectors/ebpf.plugin/README.md:200]: Actually a comma before the "and" in a list is not necessary. At least that's what my 9th grade English teacher used to insist 😄

Comment (UmanShahzad) [reply collectors/ebpf.plugin/README.md:200]: Alright well seems like it's "optional": https://www.grammarly.com/blog/comma-before-and/

Comment (vlvkobal) [reply collectors/ebpf.plugin/README.md:200]: https://learn.netdata.cloud/contribute/style-guide#oxford-comma

File: collectors/ebpf.plugin/README.md
Comment (thiagoftsm) [reply collectors/ebpf.plugin/README.md:205]: @vlvkobal you are right, I will remove the dimension in another PR.

@underhood and @UmanShahzad  I had a long talk with @hugovalente-pm about this, and I am glad I am not the unique with this opinion, I am moving the comma.

File: collectors/ebpf.plugin/README.md
Comment (thiagoftsm) [reply collectors/ebpf.plugin/README.md:200]: @UmanShahzad I also heard this a lot not only in English classes, but for Latin languages too. :smile:

File: collectors/ebpf.plugin/README.md
Comment (thiagoftsm) [reply collectors/ebpf.plugin/README.md]: Thanks, I removed.

File: collectors/ebpf.plugin/README.md
Comment (vlvkobal) [reply collectors/ebpf.plugin/README.md]: Hmm, I didn't ask you to change the first `on`. I asked to remove the second one.

**Review by vlvkobal — APPROVED** [review]

**Review by UmanShahzad — APPROVED** [review]

**Review by hugovalente-pm — APPROVED** [review]

**Pull request merged**
